app/models/user.rb

```diff
@@ -233,7 +234,8 @@ def self.find_by_username_or_email(username_or_email)
   end
 
   def self.find_by_email(email)
-    find_by(email: Email.downcase(email))
+    user_email = UserEmail.find_by(email: Email.downcase(email))
+    user_email.user if user_email
```

Comment:
In Ruby 2.3, you can do something like this, `UserEmail.find_by(email: Email.downcase(email))&.user`